plugins/LanguagesManager/LanguagesManager.php

```diff
@@ -137,18 +128,46 @@ public function uninstall()
     }
 
     /**
+     * First check forced by request language setting,
+     * then - session language setting, then - from browser,
+     * and finally default configured setting.
+     *
      * @return string Two letters language code, eg. "fr"
      */
-    public static function getLanguageCodeForCurrentUser()
+    public static function getCurrentLanguage()
     {
-        $languageCode = self::getLanguageFromPreferences();
-        if (!API::getInstance()->isLanguageAvailable($languageCode)) {
-            $languageCode = Common::extractLanguageCodeFromBrowserLanguage(Common::getBrowserLanguage(), API::getInstance()->getAvailableLanguages());
+        $languageCodes = array(
+            Common::getRequestVar('language', ''),
+            self::getLanguageForSession(),
+            self::getLanguageCodeForCurrentUser(),
+            StaticContainer::get('Piwik\Translation\Translator')->getDefaultLanguage(),
```

Comment:
It's better to pass the `$translator` in the method parameter rather than getting it again from the container (since this code is called on every request)